File: app/src/main/java/com/rkbapps/neetflix/services/RetrofitInstance.java
package com.rkbapps.neetflix.services;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitInstance {
    private static final Retrofit retrofit = new Retrofit.Builder()
            .baseUrl(ApiData.BASE_URL)
            .addConverterFactory(GsonConverterFactory.create()).build();


    private static MovieApi movieApi = null;
    private static TvSeriesApi tvSeriesApi = null;
    private static PersonApi personApi = null;


    public static MovieApi getMovieApi() {
        if (movieApi == null) {
            movieApi = retrofit.create(MovieApi.class);
        }
        return movieApi;
    }

    public static TvSeriesApi getTvSeriesApi() {
        if (tvSeriesApi == null) {
            tvSeriesApi = retrofit.create(TvSeriesApi.class);
        }
        return tvSeriesApi;
    }

    public static PersonApi getPersonApi() {
        if(personApi == null){
            personApi = retrofit.create(PersonApi.class);
        }
        return personApi;
    }
}

File: app/src/main/java/com/rkbapps/neetflix/services/RetrofitInstance.kt
package com.rkbapps.neetflix.services

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitInstance {

    private val retrofit = Retrofit.Builder()
        .baseUrl(ApiData.BASE_URL)
        .addConverterFactory(GsonConverterFactory.create()).build()

    @JvmStatic
    var movieApi: MovieApi? = null
        get() {
            if (field == null) {
                field = retrofit.create(MovieApi::class.java)
            }
            return field
        }


    @JvmStatic
    var tvSeriesApi: TvSeriesApi? = null
        get() {
            if (field == null) {
                field = retrofit.create(TvSeriesApi::class.java)
            }
            return field
        }


    var personApi: PersonApi? = null
        get() {
            if (field == null) {
                field = retrofit.create(PersonApi::class.java)
            }
            return field
        }

}